### PR TITLE
Various multi-dt fixes and CHID test

### PR DESCRIPTION
**Pull request by anonymix007**
Part of #34158

### CODE REVIEW

**Comment by bluca** [comment]
Fails on s390x so I guess endianess issue:

```
1297s /* test_chid */
1297s src/test/test-chid.c:78: Assertion failed: expected "&chids[i] == &actual_chids[i]", but cd4a7d58-7ba7-5567-98eb-382c35b34e5b != 01e09b32-de05-56ca-b9d1-9486ad5f381d
```

**Comment by anonymix007** [comment]
So it actually doesn't work with big engian machines and the code isn't correct on those. As I don't have any BE machines and this code is mainly for EFI (which is always LE), the best I can suggest is to simply not run this test on s390x.

**Comment by bluca** [comment]
unit tests need to run everywhere, so you have to detect it at runtime and skip it

File: src/test/test-chid.c [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+
+TEST(chid) {
+        /* Results compared with output of 'fwupdtool hwids' */
+        EFI_GUID chids[CHID_TYPES_MAX];
+        chid_calculate(test_fields, chids);
+        for (size_t i = 0; i < ELEMENTSOF(chids); i++) {
+                ASSERT_EQ_EFI_GUID(&chids[i], &actual_chids[i]);
+        }
+
```
Comment (bluca) [inline src/test/test-chid.c]: extra newline

File: src/test/test-chid.c [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+}
+
+
```
Comment (bluca) [inline src/test/test-chid.c]: extra newline

**Review by bluca — APPROVED** [review]

**Comment by yuwata** [comment]
Please rebase.

**Comment by anonymix007** [comment]
Rebased cleanly, so not sure why it was neccesary.

File: src/test/test-chid.c [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+        {0x00000000, 0x0000, 0x0000, {0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00}},
+};
+
+TEST(chid) {
+        /* Results compared with output of 'fwupdtool hwids' */
+        EFI_GUID chids[CHID_TYPES_MAX];
+        chid_calculate(test_fields, chids);
+        for (size_t i = 0; i < ELEMENTSOF(chids); i++) {
+                ASSERT_EQ_EFI_GUID(&chids[i], &actual_chids[i]);
+        }
```
Comment (poettering) [inline src/test/test-chid.c]: coding style, no {} around single line if blocks

File: src/test/test-chid.c [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+#include "string-util.h"
+#include "tests.h"
+
+const char16_t *const test_fields[_CHID_SMBIOS_FIELDS_MAX] = {
+        [CHID_SMBIOS_MANUFACTURER] = u"Micro-Star International Co., Ltd.",
+        [CHID_SMBIOS_PRODUCT_NAME] = u"MS-7D70",
+        [CHID_SMBIOS_PRODUCT_SKU] = u"To be filled by O.E.M.",
+        [CHID_SMBIOS_FAMILY] = u"To be filled by O.E.M.",
+        [CHID_SMBIOS_BASEBOARD_PRODUCT] = u"MPG X670E CARBON WIFI (MS-7D70)",
+        [CHID_SMBIOS_BASEBOARD_MANUFACTURER] = u"Micro-Star International Co., Ltd.",
```
Comment (poettering) [inline src/test/test-chid.c:13]: we usually align tables like this on the `=`

**Review by yuwata — APPROVED** [review]

**Pull request merged**